crates/cheatnet/src/runtime_extensions/forge_config_extension/config.rs

```diff
@@ -19,6 +23,71 @@ pub enum BlockId {
     BlockNumber(u64),
 }
 
+impl<'de> Deserialize<'de> for BlockId {
```

Comment:
I think if we slightly changed the structure of BlockId enum, we could probably do without implementing deserialize ourselves.

Can you try if you can make something like this work?

```rust

    #[derive(Debug, Clone, Serialize, Deserialize)]
    pub enum Tag {
        #[serde(rename = "latest")]
        Latest,
    }

    #[derive(Debug, Clone, Serialize, Deserialize)]
    #[serde(rename = "block_id")]
    pub enum BlockId {
        #[serde(rename = "tag")]
        BlockTag(Tag), // this must be an enum because we only want to allow `latest` and not arbitrary string
        #[serde(rename = "hash")]
        BlockHash(Felt252),
        #[serde(rename = "number")]
        BlockNumber(u64),
    }
```

Reply:
Yeah with custom deserialize we get better error message, I think its crucial to get meaningful errors so end users know where to adjust the config